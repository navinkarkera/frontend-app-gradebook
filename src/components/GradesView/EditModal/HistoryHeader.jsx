import React from 'react';
import PropTypes from 'prop-types';

/**
 * HistoryHeader
 * simple display container for an individual history table header
 * @param {string} id - header id
 * @param {string} label - header label
 * @param {string} value - header value
 */
const HistoryHeader = ({ label, value }) => (
  <div className="row">
    <div className="col-2">{label}: </div>
    <div>{value}</div>
  </div>
);
HistoryHeader.defaultProps = {
  value: null,
};
HistoryHeader.propTypes = {
  label: PropTypes.node.isRequired,
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
};

export default HistoryHeader;
