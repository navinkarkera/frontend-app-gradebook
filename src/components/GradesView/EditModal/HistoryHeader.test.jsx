import React from 'react';
import { shallow } from 'enzyme';

import HistoryHeader from './HistoryHeader';

describe('HistoryHeader', () => {
  const props = {
    label: 'Brita',
    value: 'hydration',
  };
  describe('Component', () => {
    test('snapshot', () => {
      expect(shallow(<HistoryHeader {...props} />)).toMatchSnapshot();
    });
  });
});
