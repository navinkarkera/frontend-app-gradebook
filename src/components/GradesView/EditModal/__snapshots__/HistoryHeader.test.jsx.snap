// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HistoryHeader Component snapshot 1`] = `
<div
  className="row"
>
  <div
    className="col-2"
  >
    Brita
    : 
  </div>
  <div>
    hydration
  </div>
</div>
`;
